### ISSUE TITLE
Proper way to timeout an evaluation

### ISSUE THREAD

**Issue by mpeychev** [WolframResearch/WolframClientForPython#29]
Hello,

I have been running the following code:

```py
import time

from wolframclient.evaluation import WolframLanguageSession
from wolframclient.language import wl

if __name__ == '__main__':
    session = WolframLanguageSession('/Applications/Mathematica.app/Contents/MacOS/WolframKernel')

    print('start now')
    start = time.time()
    result = session.evaluate(wl.Pause(10), timeout=2.0)
    print('end after: {:.2f} seconds'.format(time.time() - start))
    print('result is:', result)
    print()
    print('start now')
    start = time.time()
    result = session.evaluate(wl.TimeConstrained(wl.Pause(10), 2))
    print('end after: {:.2f} seconds'.format(time.time() - start))
    print('result is:', result)

    session.terminate()
```

which results in the following output:
```
start now
end after: 14.29 seconds
result is: None

start now
end after: 2.00 seconds
result is: $Aborted
```

This makes me think that the `timeout` argument of `session.evaluate(...)` does not work as expected, so I will assume that using `TimeConstrained` is the recommended way to set a timeout to an evaluation, is that right?

Mathematica® version: 12.2
OS version: macOS Big Sur 11.2.3

As a side question, is there any way to "kill" (stop/terminate) a long-running evaluation while keeping the corresponding kernel alive and ready to accept further evaluations?
If no, is it possible to "kill" (stop/terminate) a long-running evaluation together with its kernel?

I tried to dig in a bit in the code but I believe getting a reply directly from the developers will be more reliable and precise. Thank you.

**Comment by DbxDev** [comment]
That's correct. There is currently no way to interrupt, from python, the kernel evaluation, while preserving the kernel state.
Using `TimeConstrained` is the correct way to limit the execution time for a given computation.
The python side timeout is only impacting the python side code which will only wait for a result until the timeout is reached. It does not interrupt the current evaluation as you have noticed.
When the fix for #30 gets merged, one will be able to terminate a session even though the underlying kernel is busy.

**Comment by mpeychev** [comment]
Thank you for the clarification. Closing this issue as I expect the progress on this problem (e.g. future pull requests, etc.) to be tracked in #30 .
